### PR TITLE
Fix for incorrect data format being sent to API when missing documents are selected

### PR DESCRIPTION
**Pull request by james-hill-kc**
refactoring of validation controller postCheckAndConfirm to ensure correct format of data posted to backend

## Describe your changes

Moved code used to build Reason data object outside postCheckAndConfirm into its own method, refactored for clarity and extensibility.

NOTE: this also includes a find-and-replace fix for the typo "inflamatory" - @mayya-bond-kc tagging you for visibility.

## Issue ticket number and link

?

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

### CODE REVIEW

File: apps/web/src/server/app/validation/validation.controller.js [hunk 1]
```diff
@@ -354,6 +354,59 @@ export function getCheckAndConfirm(request, response) {
...
+ * @returns {object} - resulting Reasons object in valid format (see /api-docs/#/default/post_validation__id_ for details)
+ */
+function formatOutcomeReasonsDataForApiPost(outcomeType, appealDetails) {
+	let data = {};
+
+	switch (outcomeType) {
+		case 'invalid':
+			data = {
+				// eslint-disable-next-line unicorn/no-array-reduce, unicorn/prefer-object-from-entries
+				...appealDetails.invalidReasons.reduce((previous, current) => {
```
Comment (brpins) [inline apps/web/src/server/app/validation/validation.controller.js:371]: Minor thing as there's no technical impact, but this naming is a bit confusing as it doesn't really match the signature of reduce. By its name, "current" would suggest a new representation of "previous", whereas it is in fact a key for the reason, while "previous" is the map being reduced.

Comment (james-hill-kc) [reply apps/web/src/server/app/validation/validation.controller.js]: I agree these could have been clearer (this is the terminology used in the reducer examples on MDN which is why I renamed the original single-letter variables to these). I've updated to "accumulator" and "value" now.

**Review by brpins — DISMISSED** [review]

**Review by mayya-bond-kc — APPROVED** [review]

**Pull request merged**